build: migrate off of the long outdated gnome-common infra

alex-ter opened the pull request:
According to https://wiki.gnome.org/Projects/GnomeCommon/Migration
we should migrate off using gnome-common macros in our build files.

@pwithnall if you are getting GitHub emails, I'd be grateful for a comment to make sure I've done it right as it took me quite some research to come up with that, due to the fact that most of these changes took place a while ago everywhere but in Tomboy.

pwithnall commented on Makefile.am:
```diff
@@ -15,20 +14,16 @@ EXTRA_DIST = 			\
 	tomboy.spec			\
 	\
-	MAINTAINERS
+	tomboy.doap
```
I would add this line in a separate commit, since changing from `MAINTAINERS` to `*.doap` is technically unrelated to porting away from `gnome-common`.

pwithnall commented:
I can’t see anything obviously wrong with this. If `make distcheck` passes, it should be OK to push.

Separately, I noticed you’re still using `AM_GLIB_GNU_GETTEXT` and `IT_PROG_INTLTOOL` — they’re deprecated in favour of [upstream gettext](https://www.gnu.org/software/gettext/). You should consider porting to that separately.

alex-ter commented:
Thank you! Yep, the doap line just slipped in and was not meant to be committed along with this one, I just left it there for the purpose of the review. And yes, I read about gettext in the migration guide and plan to fix that as the next step - thanks for the comment.

alex-ter commented:
I've split out an unrelated `MAINTAINERS-`>`tomboy.doap` change, corrected a typo in `AC_CONFIG_LOCAL_DIR` (was `DIRS`) and tested `distcheck` to pass ok on Fedora 24, OpenSUSE 42.1, OpenSUSE Tumbleweed and Debian 8.5, so I'm reasonably sure this is ok to go.
